packages/expo-video/android/src/main/java/expo/modules/video/player/VideoPlayerSubtitles.kt

```diff
@@ -0,0 +1,127 @@
+package expo.modules.video.player
+
+import androidx.annotation.OptIn
+import androidx.media3.common.C
+import androidx.media3.common.Format
+import androidx.media3.common.MimeTypes
+import androidx.media3.common.TrackGroup
+import androidx.media3.common.TrackSelectionOverride
+import androidx.media3.common.TrackSelectionParameters
+import androidx.media3.common.Tracks
+import androidx.media3.common.util.UnstableApi
+import expo.modules.video.records.SubtitleTrack
+import java.lang.ref.WeakReference
+
+@OptIn(UnstableApi::class)
+class VideoPlayerSubtitles(owner: VideoPlayer) : VideoPlayerListener {
+  private val owner = WeakReference(owner)
+  private val videoPlayer: VideoPlayer?
+    get() {
+      return owner.get()
+    }
+  private var formatsToGroups: MutableMap<Format, Pair<TrackGroup, Int>> = mutableMapOf()
+  private var currentSubtitleFormat: Format? = null
+  private var currentOverride: TrackSelectionOverride? = null
+
+  var currentSubtitleTrack: SubtitleTrack?
+    get() {
+      return SubtitleTrack.fromFormat(currentSubtitleFormat)
+    }
+    set(value) {
+      applySubtitleTrack(value)
+    }
+  var availableSubtitleTracks: ArrayList<SubtitleTrack> = arrayListOf()
+
+  init {
+    owner.addListener(this)
+  }
+
+  fun setSubtitlesEnabled(enabled: Boolean) {
+    val currentParams = videoPlayer?.player?.trackSelectionParameters ?: return
+    var params = currentParams.buildUpon().setTrackTypeDisabled(C.TRACK_TYPE_TEXT, !enabled).build()
+    if (!enabled) {
+      params = params.buildUpon().clearOverridesOfType(C.TRACK_TYPE_TEXT).build()
+    }
+    videoPlayer?.player?.trackSelectionParameters = params
```

Comment:
```suggestion
    val player = videoPlayer?.player ?: return
    val currentParams = player.trackSelectionParameters
    var params = currentParams.buildUpon().setTrackTypeDisabled(C.TRACK_TYPE_TEXT, !enabled).build()
    if (!enabled) {
      params = params.buildUpon().clearOverridesOfType(C.TRACK_TYPE_TEXT).build()
    }
    player.trackSelectionParameters = params
```

Also, it looks like the `currentParams.buildUpon().setTrackTypeDisabled(C.TRACK_TYPE_TEXT, !enabled).build()
` is removed by `clearOverridesOfType` anyway, so I don't know if we need to call it if enabled = false. So maybe something like this would be enough? 
```suggestion
    val player = videoPlayer?.player ?: return
    val currentParams = player.trackSelectionParameters
    val params = if (enabled) {
      currentParams.buildUpon().setTrackTypeDisabled(C.TRACK_TYPE_TEXT, true).build()
    } else {
	    currentParams.buildUpon().clearOverridesOfType(C.TRACK_TYPE_TEXT).build()
    }
    player.trackSelectionParameters = params
```

Reply:
I think this won't work, the track being enabled/disabled is separate from the overrides, so with this method we wouldn't be able to re-enable it